modules/objdetect/test/test_charucodetection.cpp

```diff
@@ -99,81 +103,74 @@ void CV_CharucoDetection::run(int) {
     Size imgSize(500, 500);
     aruco::DetectorParameters params;
     params.minDistanceToBorder = 3;
-    aruco::CharucoBoard board(Size(4, 4), 0.03f, 0.015f, aruco::getPredefinedDictionary(aruco::DICT_6X6_250));
-    aruco::CharucoDetector detector(board, aruco::CharucoParameters(), params);
 
     cameraMatrix.at<double>(0, 0) = cameraMatrix.at<double>(1, 1) = 600;
     cameraMatrix.at<double>(0, 2) = imgSize.width / 2;
     cameraMatrix.at<double>(1, 2) = imgSize.height / 2;
 
     Mat distCoeffs(5, 1, CV_64FC1, Scalar::all(0));
 
-    // for different perspectives
-    for(double distance = 0.2; distance <= 0.4; distance += 0.2) {
-        for(int yaw = -55; yaw <= 50; yaw += 25) {
-            for(int pitch = -55; pitch <= 50; pitch += 25) {
+    for(bool BBF : {true, false}) {
```

Comment:
Avoid `for loops` over parameters in test bodies.
In case of failure there would be unclear message without information about used parameters.
Also due to `for loops` it is hard to debug/reproduce the problem.

It is better to add parameter and/or separate test case.